salt/haproxy/config/haproxy.cfg.jinja

```diff
@@ -117,6 +117,15 @@ frontend main
     bind :::80
     bind 127.0.0.1:19001  # This is our TLS socket.
 
+    # Client rate limiting
+    # See: https://www.haproxy.com/blog/four-examples-of-haproxy-rate-limiting
+    stick-table type ip size 100k expire 15s store http_req_rate(10s)
+    http-request track-sc0 src
+    http-request deny deny_status 429 if { sc_http_req_rate(0) gt 20 }
```

Comment:
nice! this is a really promising looking configuration. I wonder if this could be done on a service by service basis (i.e. bugs has a 10 rps limit and docs has a 100 rps).

what might that look like?

Reply:
im not sure i know the answer to this yet... maybe it would revolve around acls for different services but im not sure how that would work without digging further into docs.

my first idea is a stick-table per service we want limited, and maybe a general one that is suitable for *most* ?